server/services/location.js

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import GOOGLE_API_KEY from '../config/development';
+
+const SEARCH_RADIUS_IN_METERS = 500;
+
+module.exports = class LocationService {
+    constructor() {}
+
+    async isNearAStore() {
+        const userLocation = await this.getUserLocation();
+        if (!userLocation){
+            return false;
+        }
+        const nearbyStores = await this.findNearbyStores(userLocation.lat, userLocation.lng);
+        return nearbyStores && nearbyStores.length > 0;
+    }
+
+    async getUserLocation() {
+        try {
+            const response = await axios.post('https://www.googleapis.com/geolocation/v1/geolocate?key=' + GOOGLE_API_KEY);
+            return response.data;
+        }
+        catch (error) {
+            console.log(error);
+            return null;
+        }
+    }
+
+    async findNearbyStores(latitude, longitude){
+        const nearbyGroceryStoresUrl = `https://maps.googleapis.com/maps/api/place/findplacefromtext/json?key=${GOOGLE_API_KEY}&inputtype=textquery&input=grocery&fields=name,types&locationbias=circle:${SEARCH_RADIUS_IN_METERS}@${latitude},${longitude}`;
+
+        try {
+            let validResults = [];
+
+            const response = await axios.post(nearbyGroceryStoresUrl);
+
+            // this is the status of the actual http response
+            if (response.status !== 200){
+                console.log("Status: " + response.status);
+                return validResults;
+            }
+
+            // this is the google api's status field
+            if (response.data.status !== "OK"){
+                console.log("Status: " + response.data.status);
```

Comment:
I think there might be a better way to handle these failures besides just logging the status, but in the interest of time it's fine. But could we add something to distinguish the two messages so the origin of the non-200 is more clear?

```suggestion
                console.log("Map Status: " + response.data.status);
```